static/app/utils.tsx

```diff
@@ -120,6 +120,56 @@ export function explodeSlug(slug: string): string {
   return trim(slug.replace(/[-_]+/g, ' '));
 }
 
+/**
+ * Trim slug name with a preference for preserving whole words. Only cut up
+ * whole words if the last remaining words are still too long. For example:
+ * "javascript-project-backend" --> "javascript…backend"
+ * "my-long-sentry-project-name" --> "my-long-sentry…name"
+ * "javascriptproject-backend" --> "javascriptproje…ckend"
+ */
+export function trimSlug(slug: string, maxLength: number = 20) {
```

Comment:
I think we can probably call this just `trimMiddle`

Reply:
Oh I guess it is a bit special though, that it splits on `-`s

Reply:
Can we write some unit tests for this?

Reply:
Also let's move it to it's own file `static/app/utils/trimSlug.tsx`

Reply:
Ahh good points! Updated in https://github.com/getsentry/sentry/pull/33002/commits/254da1f7ffd990b862ae30ad54cb0834e14657ed